core/src/main/java/org/apache/spark/QueryContextSummary.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark;
+
+import org.apache.spark.annotation.Evolving;
+
+/**
+ * Build textual summary from a query context.
+ *
+ * @since 3.4.0
+ */
+@Evolving
+public interface QueryContextSummary extends QueryContext {
```

Comment:
does this need to be a public API?

Reply:
Made it as a private one.

Reply:
shall we use `private[spark]`? to make sure it won't show up in the API doc

Reply:
This is Java. Does such syntax work here?

Reply:
Since it's for internal use only, we can write it in Scala.

Reply:
+1, I don't think we need this interface. Having SQLQueryContext is enough.

Reply:
Let me try to remove it ...